server/testsupport/createGoalDtoTestDataBuilder.ts

```diff
@@ -9,6 +9,7 @@ const aValidCreateGoalDtoWithOneStep = (title = 'Learn Spanish'): CreateGoalDto
     prisonNumber: 'A1234BC',
     title,
     steps: [addStepDto],
+    targetCompletionDate: '2024-01-01',
```

Comment:
The values in your test data builders here will need to be dates rather than strings
Have a look where we've used `moment` elsewhere, but I think you want something like
```
moment('2024-01-01').toDate
```
Something like that anyway